### PR TITLE
Fix issue #215

### PR DESCRIPTION
**Pull request by jotsetung**
- Remove check for parameters being of length > 0.
- Add unit tests to ensure empty spectra are correctly returned.

### CODE REVIEW

File: R/functions-Spectrum1.R [hunk 1]
```diff
@@ -93,12 +93,15 @@ Spectra1_mz_sorted <- function(peaksCount = NULL, rt = numeric(),
...
-        stop("Arguments 'mz', 'intensity' and 'nvalues' are required!")
-    } else {
-        if (length(mz) != length(intensity))
-            stop("Lengths of 'mz' and 'intensity' do not match!")
-    }
+    ## Fix issue #215: remove check to allow empty spectra
+    ## if (length(mz) == 0 | length(intensity) == 0 | length(nvalues) == 0) {
+    ##     stop("Arguments 'mz', 'intensity' and 'nvalues' are required!")
+    ## }
+    if (sum(nvalues) != length(mz))
```
Comment (lgatto) [inline R/functions-Spectrum1.R]: Is this really `sum(nvlaues)`? Shouldn't it be `nvalues`?

**Comment by lgatto** [comment]
I would also suggest to add tests to `test_Spectrum.R` for example, where I test that an empty spectrum (levels 1 and 2) are valid. 

```r
test_that("Spectrum validity", {
  expect_true(validObject(new("Spectrum1")))
  expect_true(validObject(new("Spectrum2")))
  expect_true(validObject(sp3 <- new("Spectrum2", msLevel = 3L)))
  expect_equal(msLevel(sp3), 3L)
})
```

File: R/functions-Spectrum1.R (hunk 1 above)
Comment (jotsetung) [inline R/functions-Spectrum1.R]: `nvalues` is an `integer` of length equal to the number of spectra, each value representing the number of peaks it contains (and hence how many `mz` and `intensity` values it contains). This is used to *split* the `mz` and `intensity` vectors (that contain the mz and intensity values of all spectra) per spectrum. 
The test is thus OK.

Comment (lgatto) [inline R/functions-Spectrum1.R]: ok

File: tests/testthat/test_Spectrum_Constructor.R [hunk 2]
```diff
@@ -126,6 +128,30 @@ test_that("C-level multi-Spectrum1 constructor with M/Z ordering", {
         expect_identical(mz(spectL[[i]]), mzValsList[[i]][idxList[[i]]])
         expect_identical(intensity(spectL[[i]]), intValsList[[i]][idxList[[i]]])
     }
+
+    ## Check empty spectra.
+    ## Spectrum1
+    mzVals <- sort(abs(rnorm(200, mean = 100, sd = 10)))
```
Comment (sgibb) [inline tests/testthat/test_Spectrum_Constructor.R]: Should we use `set.seed` here? Otherwise if a test randomly fails we would never able to reproduce it because we don't know the input values.

Comment (jotsetung) [inline tests/testthat/test_Spectrum_Constructor.R]: I don't believe that these tests will ever fail (fingers crossed), but I'll add the `set.seed` to be on the save side.

Comment (lgatto) [inline tests/testthat/test_Spectrum_Constructor.R]: > I don't believe that these tests will ever fail

LOL!
